src/pages/groupComparison/Survival.tsx

```diff
@@ -0,0 +1,102 @@
+                        <SurvivalChart
+                            className='borderedChart'
+                            patientSurvivals = {this.props.store.overallPatientSurvivals.result}
+                            analysisGroups={this.props.store.sampleGroups.result!}
+                            patientToAnalysisGroup={this.props.store.patientToAnalysisGroup.result!}
+                            title={this.overallSurvivalTitleText}
+                            xAxisLabel="Months Survival"
+                            yAxisLabel="Overall Survival"
+                            totalCasesHeader="Number of Cases, Total"
+                            statusCasesHeader="Number of Cases, Deceased"
+                            medianMonthsHeader="Median Months Survival"
+                            yLabelTooltip="Survival estimate"
+                            xLabelWithEventTooltip="Time of death"
+                            xLabelWithoutEventTooltip="Time of last observation"
+                            fileName="Overall_Survival" />
+                    </div>
+                </div>
+            );
+        } else {
+            overallNotAvailable = true;
+        }
+
+        if (this.props.store.diseaseFreePatientSurvivals.isComplete &&
+            this.props.store.diseaseFreePatientSurvivals.result.length > 0 &&
+            this.props.store.sampleGroups.isComplete &&
+            this.props.store.patientToAnalysisGroup.isComplete) {
+            content.push(
+                <div>
+                    <h4 className='forceHeaderStyle h4'>{ this.diseaseFreeSurvivalTitleText }</h4>
+                    <div style={{width: '920px'}}>
+                        <SurvivalChart
+                            className='borderedChart'
+                            patientSurvivals = {this.props.store.diseaseFreePatientSurvivals.result}
+                            analysisGroups={this.props.store.sampleGroups.result!}
+                            patientToAnalysisGroup={this.props.store.patientToAnalysisGroup.result!}
+                            title={this.diseaseFreeSurvivalTitleText}
+                            xAxisLabel="Months Disease/Progression-free"
+                            yAxisLabel="Disease/Progression-free Survival"
+                            totalCasesHeader="Number of Cases, Total"
+                            statusCasesHeader="Number of Cases, Relapsed/Progressed"
+                            medianMonthsHeader="Median Months Disease-free"
+                            yLabelTooltip="Disease-free Estimate"
+                            xLabelWithEventTooltip="Time of Relapse"
+                            xLabelWithoutEventTooltip="Time of Last Observation"
+                            fileName="Disease_Free_Survival" />
+                    </div>
+                </div>
+            );
+        } else {
+            diseaseFreeNotAvailable = true;
+        }
+
+        if (overallNotAvailable && diseaseFreeNotAvailable) {
```

Comment:
wouldnt you want to split these cases up?

Reply:
or put the `push` statements instead of where you set `..NotAvailable = true`

Reply:
I was using same code from survivalTab from resultView. I'll split it